02-peer-protocol.md

```diff
@@ -544,10 +544,10 @@ A sending node:
   - if it hasn't sent a `funding_created` (if it is a funder) or a `funding_signed` (if it is a fundee):
     - MUST NOT send a `shutdown`
   - MAY send a `shutdown` before a `funding_locked`, i.e. before the funding transaction has reached `minimum_depth`.
-  - if there are updates pending on the receiving node's commitment transaction:
+  - if there are uncommitted updates pending on the receiving node's commitment transaction:
```

Comment:
@vincenzopalazzo started looking at this in our code and I kinda realized....why do we have this requirement at all? Like, who really cares what's pending when a shutdown goes out? *after* shutdown goes out you shouldn't add anything new, and certainly sending a shutdown out while an update_add is pending is kinda dumb (the peer should reject at that point), but its not like it should break anything, only pre-closing_signed is a real concern here.

Reply:
I assume a part of the reason for this is so that the receiving end can enforce the "no new update_adds" requirement when receiving `commitment_signed` and not later? Does anyone do this (maybe c-lightning @rustyrussell ?)

Reply:
OK, this was good chatting, and I'm sorry if this came out of my head after the meeting, but I just drafted the lnprototest PR https://github.com/rustyrussell/lnprototest/pull/37 after the meeting and I was able to make some consideration after writing the code.

So, my idea that I shared with @TheBlueMatt is why we care about what we don't receive, and we can be more precise in the fact of what we must not receive.

This will make the spec also less ambiguous, for instance, some things like that
```suggestion
  - if `add_htlc` message is received from the received after shutdown, the sender must fail the htlc 
```

P.S: this make also the lnprototest integration test easy with a single assert `MustNodeReceive('add_htlc')` :)

Reply:
The only reason to have a "black list" is that during the time this black list can blow up?